data-prepper-core/src/main/java/com/amazon/dataprepper/peerforwarder/PeerForwarderClientFactory.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.peerforwarder;
+
+import com.amazon.dataprepper.peerforwarder.certificate.CertificateProviderFactory;
+import com.amazon.dataprepper.peerforwarder.discovery.DiscoveryMode;
+import com.amazon.dataprepper.peerforwarder.discovery.PeerListProvider;
+
+public class PeerForwarderClientFactory {
+    public static final int NUM_VIRTUAL_NODES = 128;
+
+    private final PeerForwarderConfiguration peerForwarderConfiguration;
+    private final PeerClientPool peerClientPool;
+
+    public PeerForwarderClientFactory(PeerForwarderConfiguration peerForwarderConfiguration,
+                                      PeerClientPool peerClientPool) {
+        this.peerForwarderConfiguration = peerForwarderConfiguration;
+        this.peerClientPool = peerClientPool;
+    }
+
+    public HashRing createHashRing() {
+        final DiscoveryMode discoveryMode = peerForwarderConfiguration.getDiscoveryMode();
+        final PeerListProvider peerListProvider = discoveryMode.create(peerForwarderConfiguration);
+        return new HashRing(peerListProvider, NUM_VIRTUAL_NODES);
+    }
+
+    public PeerClientPool createPeerClientPool() {
```

Comment:
I agree, I can make this method create it's own instance instead of injecting a new instance. I just used Spring DI without creating new instance, or we can even changethe method name to `setPeerClientPool()`.
```
public PeerClientPool createPeerClientPool() {
        peerClientPool = new PeerClientPool();
```

Right now this method isn't used, but this will be used by `PeerForwarderClient` in the upcoming PR.